src/App.js

```diff
@@ -1,41 +1,77 @@
-  const urlIngredients = `https://api.spoonacular.com/recipes/findByIngredients?apiKey=${REACT_APP_API_KEY}&ingredients=${ingredients1[randomNumber1]},+${ingredients2[randomNumber2]}&number=1`;
+  let urlIngredients = '';
 
-  function handleRandomize() {
+  function onSubmitIngredients(arr) {
     async function getNewRecipe() {
       try {
+        // Check which URL should be send depending und number of ingredients
+        if (arr.length === 5) {
+          urlIngredients = `https://api.spoonacular.com/recipes/findByIngredients?apiKey=${REACT_APP_API_KEY}&ingredients=${arr[0]},+${arr[1]},+${arr[2]},+${arr[3]},+${arr[4]}&number=5`;
+        } else if (arr.length === 4) {
+          urlIngredients = `https://api.spoonacular.com/recipes/findByIngredients?apiKey=${REACT_APP_API_KEY}&ingredients=${arr[0]},+${arr[1]},+${arr[2]},+${arr[3]}&number=5`;
+        } else if (arr.length === 3) {
+          urlIngredients = `https://api.spoonacular.com/recipes/findByIngredients?apiKey=${REACT_APP_API_KEY}&ingredients=${arr[0]},+${arr[1]},+${arr[2]}&number=5`;
+        } else if (arr.length === 2) {
+          urlIngredients = `https://api.spoonacular.com/recipes/findByIngredients?apiKey=${REACT_APP_API_KEY}&ingredients=${arr[0]},+${arr[1]}&number=5`;
+        } else {
+          console.log('Something went wrong!');
+        }
+        console.log(urlIngredients);
         const response = await fetch(urlIngredients);
         const data = await response.json();
-        const recipeID = data[0].id;
-        console.log(recipeID);
-        let urlID = `https://api.spoonacular.com/recipes/${recipeID}/information?apiKey=${REACT_APP_API_KEY}`;
-        const response2 = await fetch(urlID);
-        const data2 = await response2.json();
-        console.log(data2);
-        setRecipes([...recipes, data2]);
+        console.log(data);
+        //  FAILED FETCH / if invalid ingredients where send, API responds with an empty array.
+        if (data.length === 0) {
+          setRecipes([
+            ...recipes,
+            {
+              title: 'Sorry there was no recipe found for you!',
+              image: 'https://source.unsplash.com/random/?food',
+              summary: 'Please fill in some other ingedients!',
+              url: 'https://www.google.com',
+            },
+          ]);
+          setMissingIngredients([{ name: 'Everything' }]);
+        } else {
+          let randomNumber = Math.floor(Math.random() * 5);
+          setMissingIngredients(data[randomNumber].missedIngredients);
+          let urlID = `https://api.spoonacular.com/recipes/${data[randomNumber].id}/information?apiKey=${REACT_APP_API_KEY}`;
+          const response2 = await fetch(urlID);
+          const data2 = await response2.json();
+          console.log(data2);
+          setRecipes([...recipes, data2]);
+        }
       } catch (error) {
         console.log(error);
       }
     }
     getNewRecipe();
   }
-
+  console.log(recipes);
```

Comment:
Do you still need your console.logs? 🙂

Reply:
yeah! I will rewrite the fetch in a later user story. so I'll need them.